lib/maths/time_series/CTimeSeriesDecomposition.cc

```diff
@@ -127,59 +121,38 @@ bool CTimeSeriesDecomposition::acceptRestoreTraverser(const common::SDistributio
-                        std::bind(&CSeasonalityTest::acceptRestoreTraverser,
-                                  &m_SeasonalityTest, std::placeholders::_1)))
+                    traverser.traverseSubLevel([this](auto& traverser_) {
+                        return m_SeasonalityTest.acceptRestoreTraverser(traverser_);
+                    }))
             RESTORE(CALENDAR_CYCLIC_TEST_7_11_TAG,
-                    traverser.traverseSubLevel(
-                        std::bind(&CCalendarTest::acceptRestoreTraverser,
-                                  &m_CalendarCyclicTest, std::placeholders::_1)))
-            RESTORE(COMPONENTS_7_11_TAG,
-                    traverser.traverseSubLevel(std::bind(
-                        &CComponents::acceptRestoreTraverser, &m_Components,
-                        std::cref(params), m_LastValueTime, std::placeholders::_1)))
+                    traverser.traverseSubLevel([this](auto& traverser_) {
+                        return m_CalendarCyclicTest.acceptRestoreTraverser(traverser_);
+                    }))
+            RESTORE(COMPONENTS_7_11_TAG, traverser.traverseSubLevel([&](auto& traverser_) {
+                return m_Components.acceptRestoreTraverser(params, traverser_);
+            }))
         }
     } else if (traverser.name() == VERSION_6_3_TAG) {
         while (traverser.next()) {
             const std::string& name{traverser.name()};
             RESTORE_BUILT_IN(TIME_SHIFT_6_3_TAG, m_TimeShift)
             RESTORE_BUILT_IN(LAST_VALUE_TIME_6_3_TAG, m_LastValueTime)
             RESTORE_BUILT_IN(LAST_PROPAGATION_TIME_6_3_TAG, m_LastPropagationTime)
-            RESTORE(SEASONALITY_TEST_6_3_TAG,
-                    traverser.traverseSubLevel(
-                        std::bind(&CSeasonalityTest::acceptRestoreTraverser,
-                                  &m_SeasonalityTest, std::placeholders::_1)))
+            RESTORE(SEASONALITY_TEST_6_3_TAG, traverser.traverseSubLevel([this](auto& traverser_) {
+                return m_SeasonalityTest.acceptRestoreTraverser(traverser_);
+            }))
             RESTORE(CALENDAR_CYCLIC_TEST_6_3_TAG,
-                    traverser.traverseSubLevel(
-                        std::bind(&CCalendarTest::acceptRestoreTraverser,
-                                  &m_CalendarCyclicTest, std::placeholders::_1)))
-            RESTORE(COMPONENTS_6_3_TAG,
-                    traverser.traverseSubLevel(std::bind(
-                        &CComponents::acceptRestoreTraverser, &m_Components,
-                        std::cref(params), m_LastValueTime, std::placeholders::_1)))
+                    traverser.traverseSubLevel([this](auto& traverser_) {
+                        return m_CalendarCyclicTest.acceptRestoreTraverser(traverser_);
+                    }))
+            RESTORE(COMPONENTS_6_3_TAG, traverser.traverseSubLevel([&](auto& traverser_) {
+                return m_Components.acceptRestoreTraverser(params, traverser_);
+            }))
         }
-    } else {
```

Comment:
Maybe return false if neither supported version tag is found?

Reply:
Done in [ca10699](https://github.com/elastic/ml-cpp/pull/2221/commits/ca106994958c0af9ea88d7b2bb9d008293d1281c).